src/TokenParser/FromTokenParser.php

```diff
@@ -50,11 +50,11 @@ public function parse(Token $token): Node
 
         $stream->expect(Token::BLOCK_END_TYPE);
 
-        $var = new AssignNameExpression($this->parser->getVarName(), $token->getLine());
```

Comment:
Using `AssignNameExpression` here is wrong: first, we're not storing in `$context`, but more importantly, we never compile this node, it's only used for its `name` attribute.